tests/unit_tests/test_analytics/test_sentry.py

```diff
@@ -0,0 +1,370 @@
+import os
+import socket
+import threading
+import time
+import zlib
+
+import flask
+import pytest
+import requests
+import wandb
+import wandb.analytics
+import wandb.env
+from flask import request
+from sentry_sdk.envelope import Envelope
+from sentry_sdk.utils import sentry_sdk
+
+EXTERNAL_SENTRY_DSN_KEY = "EXTERNALKEY"
+EXTERNAL_SENTRY_PROJECT = "123456"
+INTERNAL_SENTRY_DSN_KEY = "INTERNALKEY"
+INTERNAL_SENTRY_PROJECT = "654321"
+CLIENT_MESSAGE = "client message"
+WANDB_MESSAGE = "wandb message"
+
+
+def _get_free_port() -> int:
+    sock = socket.socket()
+    sock.bind(("", 0))
+    _, port = sock.getsockname()
+    return port
+
+
+PORT = _get_free_port()
```

Comment:
this is not ideal, that we could bind function just on import... could you potentially move it closer to when it is actually used?

Reply:
Good point, move this to a pytest fixture to avoid needing to spin up an flask server for each test. Which also avoids binding a port on import.